services/src/functional-tests/src/test/groovy/org/openkilda/functionaltests/spec/multitable/MultitableFlowsSpec.groovy

```diff
@@ -0,0 +1,1354 @@
+                extraRule[0].match.inPort == flow.source.portNumber.toString()
+                extraRuleOnInvolvedSwitches.put(involvedSwitches[0].dpId, extraRule[0].cookie)
+            }
+            verifyAll(northbound.getSwitchRules(involvedSwitches[1].dpId).flowEntries) { currentRules ->
+                currentRules.size() == rulesFromInvolvedSwitches[involvedSwitches[1].dpId].size() + 2
+                currentRules.findAll { Cookie.isIngressRulePassThrough(it.cookie) }.empty
+            }
+
+            verifyAll(northbound.getSwitchRules(involvedSwitches[2].dpId).flowEntries) { currentRules ->
+                currentRules.size() == rulesFromInvolvedSwitches[involvedSwitches[2].dpId].size() + 3
+                def extraRule = currentRules.findAll { Cookie.isIngressRulePassThrough(it.cookie) }
+                extraRule.size() == 1
+                extraRule[0].tableId == singleTableId
+                extraRule[0].instructions.goToTable == ingressRuleMultiTableId
+                extraRule[0].match.inPort == flow.destination.portNumber.toString()
+                extraRuleOnInvolvedSwitches.put(involvedSwitches[2].dpId, extraRule[0].cookie)
+            }
+        }
+
+        when: "Update switch props(multi_table: false) on all involved switches"
+        involvedSwitches.each {
+            northbound.updateSwitchProperties(it.dpId, northbound.getSwitchProperties(it.dpId).tap {
+                it.multiTable = false
+            })
+        }
+
+        then: "Extra flow rules are still exist on the src and dst switches"
+        involvedSwitches[0, -1].each { sw ->
+            assert northbound.getSwitchRules(sw.dpId).flowEntries.findAll {
+                extraRuleOnInvolvedSwitches[sw.dpId] == it.cookie
+            }.size() == 1
+        }
+
+        when: "Synchronize the flow"
+        northbound.synchronizeFlow(flow.flowId)
+
+        then: "Extra flow rules were removed from the src and dst switches"
+        Wrappers.wait(RULES_DELETION_TIME) {
+            involvedSwitches[0, -1].each { sw ->
+                assert northbound.getSwitchRules(sw.dpId).flowEntries.findAll {
+                    extraRuleOnInvolvedSwitches[sw.dpId] == it.cookie
+                }.empty
+            }
+        }
+
+        cleanup: "Revert the system to origin state"
+        flowHelper.deleteFlow(flow.flowId)
+        involvedSwitches.each { sw ->
+            northbound.updateSwitchProperties(sw.dpId, northbound.getSwitchProperties(sw.dpId)
+                    .tap { it.multiTable = initSwProps[sw.dpId].multiTable })
+            Wrappers.wait(RULES_INSTALLATION_TIME) {
+                assert northbound.getSwitchRules(sw.dpId).flowEntries*.cookie.sort() == sw.defaultCookies.sort()
+            }
+        }
+        database.resetCosts()
+    }
+
+    def "System creates extra default rules for a single switch flow(just for sending traffic to specific table)"() {
```

Comment:
Same. Need to discuss this style of tests

Reply:
removed